docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -382,6 +382,14 @@ used when compiling the standard library and overlays.
 
 ## `@_objc_non_lazy_realization`
 
+Marks a class as being non-lazily (i.e. eagerly) [realized](/docs/Lexicon.md#realization).
+
+This is used for declarations which may be statically referenced and wouldn't
+go through the normal lazy realization paths. For example, the empty array
+class must be non-lazily realized, because empty arrays are statically
+allocated. Otherwise, passing the empty array object to other code without
+triggering realization would load to a read from uninitialized memory.
```

Comment:
The problem is not uninitialized memory, but rather that most ObjC runtime APIs assume/require that classes are realized, and passing an unrealized class to them results in a crash because of that. I'd say, "Otherwise, passing the empty array object to other code without triggering realization could allow for the unrealized empty array class to be passed to ObjC runtime APIs which only operate on realized classes, resulting in a crash."